Libs/MRML/Core/vtkMRMLStorageNode.h

```diff
@@ -375,6 +375,17 @@ class VTK_MRML_EXPORT vtkMRMLStorageNode : public vtkMRMLNode
   const vtkMRMLMessageCollection *GetUserMessages() const { return this->UserMessages; }
   vtkMRMLMessageCollection *GetUserMessages() { return this->UserMessages; }
 
+  //@{
+  /// Ensures that the file name (excluding the extension) is shorter than the maximum allowed length.
+  /// If the filename is shorter than the maximum allowed length then it is returned unchanged.
+  /// If the filename is longer than the maximum allowed length then the filename is shortened by using the following format:
+  /// [first 20 characters of the base name]_[4 character hash code].[extension]
+  /// The length of the prefix is the maximum allowed length minus the length of the hash code plus one for the added underscore.
+  /// The full base name of the file will be exactly maxFileNameLength characters long.
+  std::string ClampFileName(const std::string& filename, int maxFileNameLength = 25, int hashLength = 4);
```

Comment:
It is OK to hardcode the 25 but it is error-prone to hardcode it at multiple places. Let's add a static function in the vtkMRMLStorageNode and use that everywhere to specify the filename (and maybe also do it for the hash length). Defaults could be set to `-1`, which would mean that the `value returned by vtkMRMLStorageNode::GetPreferredFileNameLength()` would be used. Maybe `Preferred` is not the best word, it would be just important to qualify what kind of maximum is this (distinguish from the file system or other hard maximum lengths).

Reply:
I've used "GetRecommendedFileNameLength" for the function name.